### ISSUE TITLE
region off-by-one when evil mode

### ISSUE THREAD

**Issue by unhammer** [oantolin/embark#478]
I first noticed this in one of my embark-keymap-alist entries, but it's reproducible with e.g. DEL on the region. If you have evil-mode and mark a region and use embark from the minibuffer, the end of the region will be one shorter compared to what it is before you hit M-o.

Here e.g. without evil it deletes the whole word, with evil it doesn't:
![region-off-by-one](https://user-images.githubusercontent.com/56868/160005104-52242834-c3a8-4110-b5d8-7395c2b79311.gif)

and the result `(buffer-substring (region-beginning) (region-end))` is actually different before/after

**Comment by oantolin** [comment]
I could not reproduce this with that version of Embark and Evil 1.1.4.0 (which I got from MELPA). I tried 6 different combinations:

Three ways of marking the word: (1) using `M-@`, (2) using visual state: `ve`, (3) double-clicking with the mouse (which I *think* is what you did in the gif).

Two ways of deleting the region: (1) `C-. DEL`, (2) `M-x C-. DEL` (which I *think* is what you did in the gif).

All 6 combinations deleted the word correctly.

**Comment by unhammer** [comment]
Well that is odd. I'll try some different emacses, maybe the bug isn't in embark at all

**Comment by oantolin** [comment]
Did you remove a comment saying the bug did not appear with Evil 1.14.0? Does that mean you can reproduce it with 1.14.0?

**Comment by unhammer** [comment]
yes and no, sorry for the noise. I first thought I could, but then I noticed that even though I can `evil-mode 1` with 1.14.0, the `require 'evil` actually fails (because of use of a deprecated form in evil-ex.el). So if I only partially load evil-mode, it works great! But if I remove that deprecated form (a `define-obsolete-function-alias` with just 2 args) so that evil fully loads, I can reproduce with 1.14.0.

**Comment by unhammer** [comment]
It works if I 

```
diff --git a/evil-types.el b/evil-types.el
index c23fd59..9cdcf29 100644
--- a/evil-types.el
+++ b/evil-types.el
@@ -88,7 +88,7 @@ If the end position is at the beginning of a line, then:
               (format "%s character%s" width
                       (if (= width 1) "" "s")))))
 
-(evil-define-type inclusive
+(evil-define-type inclusiveremoved
   "Include the character under point.
 If the end position is at the beginning of a line or the end of a
 line and `evil-want-visual-char-semi-exclusive', then:

```
but I don't know evil enough to know what that means

**Comment by oantolin** [comment]
I don't know enough about evil to now exactly what this means either.
